pkg/iam/iam.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package iam
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/minio/minio/pkg/auth"
+	"github.com/minio/minio/pkg/iam/policy"
+	"github.com/minio/minio/pkg/iam/validator"
+	xnet "github.com/minio/minio/pkg/net"
+)
+
+type iamType string
+
+// IAM types implemented.
+const (
+	IAMOpenID iamType = "openid"
+	IAMMinio  iamType = "minio"
+
+	// Add new IAM types here.
+)
+
+type policyType string
+
+// Policy types implemented.
+const (
+	PolicyOPA   policyType = "opa"
+	PolicyMinio policyType = "minio"
+)
+
+// IAM implements all definitions to capture multi user
+// and external identity providers.
+type IAM struct {
+	Version  string `json:"version"`
+	Identity struct {
+		Type   iamType `json:"type"`
+		OpenID struct {
+			JWT validator.JWTArgs `json:"jwt"`
+		} `json:"openid"`
+		Minio struct {
+			Users map[string]auth.Credentials `json:"users"`
```

Comment:
AB wanted this:
```
                "harsha": {
                    secretKey: "harshasecret",
                    status: "enabled"
                },
```

Reply:
I didn't add it because I didn't see a good reason to enabled/disabled doesn't make any sense. if needed we can add it later. Also, this structure is not good with coding changes too many things elsewhere in the code. The current behavior is more acceptable.

Reply:
You can check with AB on this because he was particular about "status" field

Reply:
Will check, we can add it anyway without disturbing the config later too.